system-configuration/src/network_configuration.rs

```diff
@@ -0,0 +1,427 @@
+use core_foundation::array::CFArray;
+use core_foundation::base::{CFType, TCFType};
+use core_foundation::base::kCFAllocatorDefault;
+use core_foundation::dictionary::CFDictionary;
+use core_foundation::string::{CFString, CFStringRef};
+
+use dynamic_store::SCDynamicStoreBuilder;
+pub use system_configuration_sys::network_configuration::*;
+use system_configuration_sys::preferences::SCPreferencesCreate;
+
+use std::{fmt, ptr};
+use std::net::IpAddr;
+
+#[derive(Debug)]
+pub struct SCNetworkInterfaceMTU {
+    cur: u32,
+    min: u32,
+    max: u32,
```

Comment:
I read in the documentation that `min` and `max` can be unavailable for some interfaces, and then a negative value is returned. I see you just cast the signed integers to unsigned ones. It would be more Rust idiomatic to represent the missing values with an `Option<u32>`. And then where you obtain them from the OS you have a safety check instead of just casting, something like:
```rust
SCNetworkInterfaceMTU {
    cur: current as u32,
    min: if min < 0 { None } else { Some(min as u32) }
    max: if max < 0 { None } else { Some(max as u32) }
}
```

Reply:
Got it.